internal/service/streamprocessor/data_source_plural.go

```diff
@@ -0,0 +1,42 @@
+package streamprocessor
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"go.mongodb.org/atlas-sdk/v20240805001/admin"
+)
+
+func (d *streamProcessorsDS) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var streamConnectionsConfig TFStreamProcessorsDSModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &streamConnectionsConfig)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := d.Client.AtlasV2
+	projectID := streamConnectionsConfig.ProjectID.ValueString()
+	instanceName := streamConnectionsConfig.InstanceName.ValueString()
+	itemsPerPage := streamConnectionsConfig.ItemsPerPage.ValueInt64Pointer()
+	pageNum := streamConnectionsConfig.PageNum.ValueInt64Pointer()
+
+	apiResp, _, err := connV2.StreamsApi.ListStreamProcessorsWithParams(ctx, &admin.ListStreamProcessorsApiParams{
```

Comment:
@AgustinBettati @EspenAlbert @oarbusi @maastha @lantoli how are we deciding on the pagination here? Didn't we decide not to expose pagination and return all the results together?

Reply:
It rings the "bell" to me that we have decided not to expose these.
But so far, I have only found [one place where we do the full iteration through all pages](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/serverlessinstance/data_source_serverless_instances.go#L68)

I can take a look into creating a generic solution

Reply:
Yes this a good point. Since we captured the intention of moving away from exposing pagination details we should align this with our 2 new plural data source implementations.

Reply:
Updated with a generic pagination.
See the `AllPages` function.
I'm unhappy about the implementation (it requires defining a clunky `listOnPage` function), but it might be good enough until we implement a better solution in the SDK.

Reply:
I added an item for us to align on this in the weekly meeting on Monday, we can include in the best practices after that.

My take would be to not expose these attributes & have the Read fetch all records available for now.
 
(one small advantage is also that these are optional attributes that require us to keep track of these attributes from the TF config in our model conversion methods, [example](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/5f715ab17979b2f2be2e0323be5ebf975da707df/internal/service/streamconnection/model_stream_connection.go#L164))

Reply:
Removed all pagination parameters in [243221c](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2505/commits/243221ca7e5a942c7a4e55c1914b5ea041655460)